### PR TITLE
OpenTelemetry testing improvements

### PR DESCRIPTION
**Pull request by ammachado**
# Description

This PR adds leaking spans detection, OTEL log reporting and prints a tree with the generated traces from the test run to help identify issues in user code.

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

**Comment by davsclaus** [comment]
When you work on PRs in the future can you avoid doing code changes in the same PRs that change comments, coding standard etc in other codes. The PRs should be easier to review and ensure other code stays the same without causing unforseen problems. You can send other PRs for "cleanup" stuff.

File: core/camel-api/src/main/java/org/apache/camel/ExchangePropertyKey.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
  */
 public enum ExchangePropertyKey {
 
+    ACTIVE_SPAN("OpenTracing.activeSpan"),
```
Comment (davsclaus) [inline core/camel-api/src/main/java/org/apache/camel/ExchangePropertyKey.java:26]: Can you avoid doing changes to core in this PR and leave this where it was before.
Adding new to Exchange should be done in another PR

File: parent/pom.xml [hunk 2]
```diff
@@ -388,6 +388,8 @@
         <opensearch-testcontainers-version>2.1.0</opensearch-testcontainers-version>
         <openstack4j-version>3.12</openstack4j-version>
         <opentelemetry-version>1.41.0</opentelemetry-version>
+        <opentelemetry-log4j2-version>2.7.0-alpha</opentelemetry-log4j2-version>
```
Comment (davsclaus) [inline parent/pom.xml:391]: Hmm this is alpha software so we may have trouble maintaining it since this code change change a lot as it would not be stable yet.

**Review by davsclaus — CHANGES_REQUESTED** [review]
Thanks for all the work on this. See my prev comments.

There are 2 concerns
- the exchange property key should not be in core (yet)
- alpha software versions

File: parent/pom.xml (hunk 2 above)
Comment (ammachado) [reply parent/pom.xml:391]: All versions available for this component are marked as `alpha`, unfortunately. This is used for testing only.

File: core/camel-api/src/main/java/org/apache/camel/ExchangePropertyKey.java (hunk 1 above)
Comment (ammachado) [reply core/camel-api/src/main/java/org/apache/camel/ExchangePropertyKey.java:26]: I will revert this change.

File: parent/pom.xml (hunk 2 above)
Comment (davsclaus) [reply parent/pom.xml:391]: okay are they released regulay ie how old is those versions you use now ?

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
This PR on CI reports test failures such as


Failed Test | Duration | Failure Type
-- | -- | --
org.apache.camel.opentelemetry.ClientRecipientListRouteTest.testRoute | 5.013s | FAILURE
org.apache.camel.opentelemetry.MulticastParallelRouteTest.testRoute | 5.023s | FAILURE
org.apache.camel.opentelemetry.MulticastRouteTest.testRoute | 4.547s | FAILURE
org.apache.camel.opentelemetry.TwoServiceWithExcludeTest.testRoute | 0.001s | ERROR
org.apache.camel.opentelemetry.OpenTelemetryPropagateContextTest.testTracingOfProcessors | 1.247s | FAILURE
org.apache.camel.opentelemetry.ABCRouteTest.testRoute | 6.668s | FAILURE
org.apache.camel.opentelemetry.CustomComponentNameRouteTest.testRoute | 6.090s | FAILURE

**Comment by ammachado** [comment]
Thanks @davsclaus for the help. The tests are fixed.

**Review by orpiske — CHANGES_REQUESTED** [review]
I'm not sure why the log4j configuration file is being replaced.

File: components/camel-opentelemetry/src/test/resources/log4j2.properties
[the remark this answers is not in the document]
Comment (davsclaus) [reply components/camel-opentelemetry/src/test/resources/log4j2.properties:1]: Yes we must keep the existing log4j properties that is what we use

**Review by orpiske — APPROVED** [review]
LGTM, thanks!

**Comment by davsclaus** [comment]
We need to do something about double closing

2024-08-21 06:24:25,911 [Pool-2-worker-1] ERROR ActiveSpanManager              - Tracing: scope already closed: io.opentelemetry.context.ThreadLocalContextStorage$ScopeImpl@21264a6a

These kind of ERROR should not happen usually and it may be that we should ignore double closing instead.

**Comment by orpiske** [comment]
I'm seeing a couple of failures in the `camel-opentelemetry`. We might need to fix this or rollback the changes.
